lib/plugin-loader.js

```diff
@@ -752,71 +782,240 @@ PluginLoader.prototype = {
+          if (plugin) {
+            bootstrapLogger.info("ZWED0124I", plugin.identifier, plugin.location); //bootstrapLogger.log(bootstrapLogger.INFO,
+              //`Plugin ${plugin.identifier} at path=${plugin.location} loaded.\n`);
+              bootstrapLogger.debug("ZWED0164I", plugin.toString()); //bootstrapLogger.debug(' Content:\n' + plugin.toString());
+            let frozen = zluxUtil.deepFreeze(plugin);
+            this.plugins.push(frozen);
+            newPlugins.push(frozen);
+            this.pluginMap[plugin.identifier] = plugin;
+            successCount++;
+          } else {
+            bootstrapLogger.info("ZWED0125I", pluginDef.identifier); //bootstrapLogger.log(bootstrapLogger.INFO,
+              //`Plugin ${pluginDef.identifier} not loaded`);
+            this.pluginMap[pluginDef.identifier] = {}; //mark it as having failed so it will not be retried
+          }
+        } catch (e) {
+          bootstrapLogger.warn("ZWED0035W", pluginDef.identifier, e);
+          //downstream will pick up error
+          newPlugins.push(Object.assign(pluginDef, {error:e}));
+        }
+      }
+      this.registerStaticPluginsWithManagers(sortedAndRejectedPlugins.plugins);
+      for (const plugin of newPlugins) {
+        this.emit('pluginFound', {
+          data: plugin,
+          count: newPlugins.length
+        });
+      }
+  },
+
+  // Note - Not to be confused with auth capabilities, that describe what an auth plugin can do
+  async getComponentCapabilities(config) {
+    let appServerComp = {};
+    let agentComp = {};
+    let host;
+    let port;
+
+    appServerComp.os = process.platform; // Operating system
+    appServerComp.cpu = process.arch; // CPU architecture
+
+    if (config && config.agent.host) {
+      host = config.agent.host;
+      if (config.agent.http) {
+        port = config.agent.http.port;
+      } else if (config.agent.https) {
+        port = config.agent.https.port;
       }
     }
-    const sortedAndRejectedPlugins = depgraph.processImports();
-    sortedAndRejectedPlugins.plugins = sortedAndRejectedPlugins.plugins.filter((plugin)=> {
-      return !this.pluginMap[plugin.identifier];
+    if (host && port) {
+      let url = 'http://' + host + ':' + port + '/server/agent/environment';
+    
+      return new Promise((resolve, reject) => { /* Obtains and stores environment information from agent */
+        http.get(url, (res) => {
+        const { statusCode } = res; // TODO: Check status code for bad status
+        const contentType = res.headers['content-type'];
+
+        res.setEncoding('utf8');
+        let rawData = '';
+        res.on('data', (chunk) => { rawData += chunk; });
+        res.on('end', () => {
+          try {
+            const parsedData = JSON.parse(rawData);
+            if (parsedData.agentVersion) {
+              agentComp.version = parsedData.agentVersion;
+            }
+            if (parsedData.arch) {
+              agentComp.cpu = parsedData.arch;
```

Comment:
Shouldn't it go to `agentComp.arch`?

Reply:
no, arch is meant to be cpu architecture